### ISSUE TITLE
list vs scalar context lost

### ISSUE THREAD

**Issue by cono** [iph0/Redis-ClusterRider#1]
Hello,

We are using Redis library for our DEV environment and ClusterRider as a production solution with just simple inheritance. And I found issue in the following code:

```
    my @fields = qw(data cryptor gzip);
    my %data;
    @data{@fields} = $self->{redis}->hmget($key, @fields);
```

and on dev environment `$self->{redis}` is instance of Redis class and in production it's Redis::ClusterRider.

But because of the following code:

`    $reply = $node->$cmd_method( @{$args} );`

We are having bug, because there is force to use scalar context, and Redis returns array reference instead of list.

I reworked our code to use scalar context for both libraries, but it would be great if Redis::ClusterRider also can account wantarray logic.

**Comment by zhmylove** [comment]
It's a real issue. 
```perl
perl -Mstrict -Mwarnings -MRedis -MRedis::ClusterRider -MData::Dumper -le '
my $r = Redis::ClusterRider->new(startup_nodes=>["localhost:7001"]); 
my $scalar = $r->keys("*"); 
my @array = $r->keys("*"); 
print Dumper [$scalar, \@array]'
$VAR1 = [
          6,
          [
            6
          ]
        ];
```
Redis base does this correctly. On the same node the result is:
```
my $r = Redis->new(server => "localhost:7001");
$VAR1 = [
          6,
          [
            'q13',
            'gr5',
            'gr1',
            'q5',
            'q9',
            'gr2'
          ]
        ];
```

**Comment by zhmylove** [comment]
I've created pull request of the fix

**Comment by iph0** [comment]
Fixed in version v0.20
